rust/datafusion/src/execution/context.rs

```diff
@@ -94,7 +96,7 @@ use crate::sql::{
 /// ```
 pub struct ExecutionContext {
     /// Internal state for the context
-    pub state: Arc<Mutex<ExecutionContextState>>,
```

Comment:
The main aim of this PR is to remove this `Arc<Mutex<>>` -- the rest of the PR is a result of that change